### PR TITLE
Change generate order in addressBook

### PR DESCRIPTION
**Pull request by adudekyeti**
Change generate order in addressBook

### CODE REVIEW

File: modules/OSSMail/models/AddressBook.php [hunk 1]
```diff
@@ -32,22 +32,31 @@ class OSSMail_AddressBook_Model
...
 				$users = explode(',', ltrim($users, ','));
 				foreach ($users as &$user) {
-					$mails[$user][] = "$name <$email>";
+					$mails[$user][$setype][] = "$name <$email>";
 				}
 			}
 		}
+		foreach ($mails as $userId => $mail) {
+			$mailsToFile[$userId] = [];
+			$mailsToFile[$userId] = array_merge($mailsToFile[$userId], $mail['Contacts']);
```
Comment (mariuszkrzaczkowski) [inline modules/OSSMail/models/AddressBook.php]:
```suggestion
			$mailsToFile[$userId] = $mail['Contacts'];
```

File: modules/OSSMail/models/AddressBook.php [hunk 2]
```diff
@@ -32,22 +32,31 @@ class OSSMail_AddressBook_Model
...
+			unset($mail['Contacts']);
```
Comment (mariuszkrzaczkowski) [inline modules/OSSMail/models/AddressBook.php]:
```suggestion
			unset($mail['OSSEmployees'],$mail['Contacts']);
```

File: modules/OSSMail/models/AddressBook.php [hunk 3]
```diff
@@ -32,22 +32,30 @@ class OSSMail_AddressBook_Model
...
 				$users = explode(',', ltrim($users, ','));
 				foreach ($users as &$user) {
-					$mails[$user][] = "$name <$email>";
+					$mails[$user][$setype][] = "$name <$email>";
 				}
 			}
 		}
+		foreach ($mails as $userId => $mail) {
+			$mailsToFile[$userId] = array_merge($mail['OSSEmployees'] ?? [], $mail['Contacts'] ?? []);
+			unset($mail['OSSEmployees'], $mail['Contacts']);
```
Comment (rskrzypczak) [inline modules/OSSMail/models/AddressBook.php]: ?

File: modules/OSSMail/models/AddressBook.php [hunk 4]
```diff
@@ -32,22 +32,30 @@ class OSSMail_AddressBook_Model
...
+			foreach ($mail as $otherMail) {
+				$mailsToFile[$userId] = array_merge($mailsToFile[$userId], $otherMail);
+			}
+		}
 		$dataReader->close();
 		$fstart = '<?php $bookMails =';
-		foreach ($mails as $user => $file) {
+		foreach ($mailsToFile as $user => $file) {
```
Comment (rskrzypczak) [inline modules/OSSMail/models/AddressBook.php]: $mailsToFile not declared.

Comment (rskrzypczak) [reply modules/OSSMail/models/AddressBook.php]: array_unique($file);

**Pull request merged**